src/common.rs

```diff
@@ -148,6 +148,94 @@ pub fn solve_quadratic(c0: f64, c1: f64, c2: f64) -> ArrayVec<[f64; 2]> {
     result
 }
 
+/// Solve an arbitrary function for a zero-crossing.
+///
+/// This uses the [ITP method], as described in the paper
+/// [An Enhancement of the Bisection Method Average Performance Preserving Minmax Optimality].
+///
+/// The values of `ya` and `yb` are given as arguments rather than
+/// computed from `f`, as the values may already be known, or they may
+/// be less expensive to compute as special cases.
+///
+/// It is assumed that `ya < 0.0` and `yb > 0.0`, otherwise unexpected
+/// results may occur.
```

Comment:
is it worth enforcing this with an assert?

Reply:
I don't think it should be an assert, because weird stuff happens sometimes and crashing would be unfriendly. The client of this function is expected to be pretty sophisticated about what they want, and the point of this comment is to set expectations.